nengo/tests/test_neurons.py

```diff
@@ -476,6 +476,95 @@ def test_frozen():
+    assert (repr(nengo.LIF(tau_rc=0.05, tau_ref=0.02)) ==
+            "LIF(tau_rc=0.05, tau_ref=0.02)")
+    assert (repr(nengo.LIF(tau_rc=0.05, amplitude=2)) ==
+            "LIF(tau_rc=0.05, amplitude=2)")
+    assert (repr(nengo.LIF(tau_ref=0.02, amplitude=2)) ==
+            "LIF(tau_ref=0.02, amplitude=2)")
+    assert (repr(nengo.LIF(tau_rc=0.05, tau_ref=0.02, amplitude=2)) ==
+            "LIF(tau_rc=0.05, tau_ref=0.02, amplitude=2)")
+    assert (repr(nengo.LIF(tau_rc=0.05, tau_ref=0.02,
+                           min_voltage=-0.5, amplitude=2)) ==
+            "LIF(tau_rc=0.05, tau_ref=0.02, min_voltage=-0.5, amplitude=2)")
+
+    assert repr(nengo.AdaptiveLIFRate()) == "AdaptiveLIFRate()"
+    assert (repr(nengo.AdaptiveLIFRate(tau_n=0.1)) ==
+            "AdaptiveLIFRate(tau_n=0.1)")
+    assert (repr(nengo.AdaptiveLIFRate(inc_n=0.5)) ==
+            "AdaptiveLIFRate(inc_n=0.5)")
+    assert (repr(nengo.AdaptiveLIFRate(tau_rc=0.1)) ==
+            "AdaptiveLIFRate(tau_rc=0.1)")
+    assert (repr(nengo.AdaptiveLIFRate(tau_ref=0.1)) ==
+            "AdaptiveLIFRate(tau_ref=0.1)")
+    assert (repr(nengo.AdaptiveLIFRate(amplitude=2)) ==
+            "AdaptiveLIFRate(amplitude=2)")
+    assert (repr(nengo.AdaptiveLIFRate(tau_n=0.1, inc_n=0.5, tau_rc=0.05,
+                                       tau_ref=0.02, amplitude=2)) ==
+            "AdaptiveLIFRate(tau_n=0.1, inc_n=0.5, tau_rc=0.05, "
+            "tau_ref=0.02, amplitude=2)")
+
+    assert repr(nengo.AdaptiveLIF()) == "AdaptiveLIF()"
+    assert repr(nengo.AdaptiveLIF(tau_n=0.1)) == "AdaptiveLIF(tau_n=0.1)"
+    assert repr(nengo.AdaptiveLIF(inc_n=0.5)) == "AdaptiveLIF(inc_n=0.5)"
+    assert repr(nengo.AdaptiveLIF(tau_rc=0.1)) == "AdaptiveLIF(tau_rc=0.1)"
+    assert repr(nengo.AdaptiveLIF(tau_ref=0.1)) == "AdaptiveLIF(tau_ref=0.1)"
+    assert (repr(nengo.AdaptiveLIF(min_voltage=-0.5)) ==
+            "AdaptiveLIF(min_voltage=-0.5)")
+    assert (repr(nengo.AdaptiveLIF(tau_n=0.1, inc_n=0.5, tau_rc=0.05,
+                                   tau_ref=0.02, min_voltage=-0.5,
+                                   amplitude=2)) ==
+            "AdaptiveLIF(tau_n=0.1, inc_n=0.5, tau_rc=0.05, tau_ref=0.02, "
+            "min_voltage=-0.5, amplitude=2)")
+
+    assert repr(nengo.Izhikevich()) == "Izhikevich()"
+    assert (repr(nengo.Izhikevich(tau_recovery=0.1)) ==
+            "Izhikevich(tau_recovery=0.1)")
+    assert (repr(nengo.Izhikevich(coupling=0.3)) ==
+            "Izhikevich(coupling=0.3)")
+    assert (repr(nengo.Izhikevich(reset_voltage=-1)) ==
+            "Izhikevich(reset_voltage=-1)")
+    assert (repr(nengo.Izhikevich(reset_recovery=5)) ==
+            "Izhikevich(reset_recovery=5)")
+    assert (repr(nengo.Izhikevich(tau_recovery=0.1, coupling=0.3,
+                                  reset_voltage=-1, reset_recovery=5)) ==
+            "Izhikevich(tau_recovery=0.1, coupling=0.3, "
+            "reset_voltage=-1, reset_recovery=5)")
```

Comment:
So the one tricky thing about this test is that it doesn't check for new unrepresented parameters, which is really what causes these issues in the first place. That said, the `strings` notebook would only help with this if people think to run the strings notebook. I think this test is definitely a good addition, but we might want to look at a more programmatic/introspective way of checking the attributes of our objects to make sure all `Parameters` get documented in the `repr` string somewhere (I'm happy to leave that as future work, though).